probe/endpoint/ebpf.go

```diff
@@ -123,32 +163,34 @@ func (t *EbpfTracker) run() {
 		}
 		destPort := uint16(dPort)
 
-		var evt event
-		switch line[0] {
-		case "connect":
-			evt = Connect
-		case "accept":
-			evt = Accept
-		case "close":
-			evt = Close
-		}
+		netns := line[6]
 
-		e := ConnectionEvent{
-			Type:          evt,
-			Pid:           pid,
-			SourceAddress: sourceAddr,
-			DestAddress:   destAddr,
-			SourcePort:    sourcePort,
-			DestPort:      destPort,
-		}
+		tuple := fourTuple{sourceAddr.String(), destAddr.String(), sourcePort, destPort}
 
-		t.Events = append(t.Events, e)
+		t.handleFlow(eventStr, tuple, pid, netns)
 	}
 }
 
-// WalkEvents - walk through the connectionEvents
-func (t EbpfTracker) WalkEvents(f func(ConnectionEvent)) {
-	for _, event := range t.Events {
-		f(event)
+// walkFlows calls f with all active flows and flows that have come and gone
+// since the last call to walkFlows
+func (t *EbpfTracker) walkFlows(f func(ebpfConnection)) {
+	t.Lock()
```

Comment:
This function is called from the main thread doing the report. It iterates over the set of connections. The set of connections is written by another go routine, the one that reads from the python.